### ISSUE TITLE
Google Play Services Library Error after upgrade targetSdkVersion 29>30

### ISSUE THREAD

**Issue by grimaldistudio** [OneSignal/OneSignal-Cordova-SDK#753]
**Description:**

After upgrading to android sdk30 one signal stopped working. 
On the OneSignal dashboard I see the error Google Play Services Library Error:
Google Play services library initialization error. Check for conflicting plugins and make sure "com.google.android.gms.version" is in your AndroidManifest.xml. 

LOGCAT: Error Getting Google Registration ID INVALID_TARGET_VERSION .... at com.google.android.gms.gcm.GoogleCloudMessaging.zza (Unknown Soruce: 101) at .... 

**Environment**

Ionic-cordova
"onesignal-cordova-plugin": "^2.3.3",
Cordova CLI       : 7.1.0
Cordova Platforms : android 6.4.0

Tested on device:
Redmi Note 8 Pro (Android11)
SM-G920F (7.0)

**Steps to Reproduce Issue:**

**Anything else:**

I have really tried everything by searching for solutions on google. I always have the same problem. Can anyone show me a valid "lead" to follow? Thank you.

**Comment by jkasten2** [comment]
@grimaldistudio I haven't seen the `INVALID_TARGET_VERSION` error before. Can you provide the full stack trace so we can confirm it is coming OneSignal?

I also recommend updating to the latest OneSignal-Cordova-SDK 3.x.x as this could also fix your issue:
https://documentation.onesignal.com/docs/mobile-2020-api-migration-guide

**Comment by grimaldistudio** [comment]
This is what I get when the oneSignal log is active: 
window.plugins.OneSignal.setLogLevel ({logLevel: 4, visualLevel: 4});

![immagine](https://user-images.githubusercontent.com/3298251/144011243-dcfd6b98-5996-43ec-9136-2180841feb60.png)


Unfortunately at the moment I cannot upgrade to OneSignal-Cordova-SDK 3.x.x because I have several cordova plugins which are not compatible with androidX

**Comment by jkasten2** [comment]
@grimaldistudio Can you try updating to the latest `onesignal-cordova-plugin` 2.x.x version? I see you are using 2.3.3 which is very old, 2.11.4 is the current latest 2.x.x version.

Could you provide a list of plugin and versions you are using in your project?

**Comment by grimaldistudio** [comment]
Yes of course, I upgraded to version 2.11.4 .
Now a different error appears 

![immagine](https://user-images.githubusercontent.com/3298251/144015445-afc8ebf9-9341-4564-bf92-7ae70826759e.png)

List of plugin:
com-badrit-base64 0.2.0 "Base64"
com-sarriaroman-photoviewer 1.1.18 "PhotoViewer"
com.lampa.startapp 6.1.6 "startApp"
com.napolitano.cordova.plugin.intent 0.1.3 "IntentPlugin"
cordova-plugin-actionsheet 2.3.3 "ActionSheet"
cordova-plugin-android-imei 1.0.0 "Android IMEI"
cordova-plugin-android-permissions 1.0.0 "Permissions"
cordova-plugin-appavailability 0.4.2 "AppAvailability"
cordova-plugin-appversion 1.0.0 "App Version"
cordova-plugin-backbutton 0.3.0 "Backbutton"
cordova-plugin-background-mode 0.7.2 "BackgroundMode"
cordova-plugin-ble-central 1.2.5 "BLE"
cordova-plugin-call-number 1.0.1 "Cordova Call Number Plugin"
cordova-plugin-camera 4.0.3 "Camera"
cordova-plugin-clear-app-data 0.0.3 "Cordova Clear Data"
cordova-plugin-device 1.1.7 "Device"
cordova-plugin-dialogs 2.0.1 "Notification"
cordova-plugin-document-scanner 4.0.0 "Scan"
cordova-plugin-document-viewer 0.9.10 "SitewaertsDocumentViewer"
cordova-plugin-doze-optimize 0.1.3 "DozeOptimize"
cordova-plugin-file 6.0.1 "File"
cordova-plugin-file-opener2 2.0.19 "File Opener2"
cordova-plugin-file-transfer 1.7.1 "File Transfer"
cordova-plugin-geolocation 4.0.1 "Geolocation"
cordova-plugin-inappbrowser 3.0.0 "InAppBrowser"
cordova-plugin-mauron85-background-geolocation 3.0.6 "CDVBackgroundGeolocation"
cordova-plugin-nativeaudio 3.0.6 "Cordova Native Audio"
cordova-plugin-screen-orientation 3.0.1 "Screen Orientation"
cordova-plugin-splashscreen 5.0.2 "Splashscreen"
cordova-plugin-statusbar 2.4.2 "StatusBar"
cordova-plugin-whitelist 1.3.3 "Whitelist"
cordova-plugin-x-toast 2.6.2 "Toast"
es6-promise-plugin 4.2.2 "Promise"
ionic-plugin-keyboard 2.2.1 "Keyboard"
onesignal-cordova-plugin 2.11.4 "OneSignal Push Notifications"
phonegap-nfc 1.2.0 "NFC"
phonegap-plugin-barcodescanner 7.0.4 "BarcodeScanner"
uk.co.workingedge.phonegap.plugin.launchnavigator 4.2.0 "Launch Navigator"

**Comment by jkasten2** [comment]
@grimaldistudio There is normally an error / warning before the "ONESIGNAL_SDK_FCM_APP_NAME already exists" error, as it most likely attempt and ran into a previous error. Can show the previous one?

Also can you use logcat and paste the text instead of a screenshot?

**Comment by grimaldistudio** [comment]
Yes, you are absolutely right better to use the logcat.

08:08.077 18827-18978/XXXXXXX W/PluginManager: THREAD WARNING: exec() call to OneSignalPush.setLogLevel blocked the main thread for 17ms. Plugin should use CordovaInterface.getThreadPool().
2021-11-30 10:08:08.110 18827-18997/XXXXXXX I/OneSignal: Restoring notifications
2021-11-30 10:08:08.112 18827-18997/XXXXXXX I/OneSignal: Querying DB for notifs to restore: created_time > 1637658488 AND dismissed = 0 AND opened = 0 AND is_summary = 0 AND expire_time > 1638263288
2021-11-30 10:08:08.140 18827-18978/XXXXXXX W/PluginManager: THREAD WARNING: exec() call to OneSignalPush.init blocked the main thread for 58ms. Plugin should use CordovaInterface.getThreadPool().
2021-11-30 10:08:08.498 18827-19011/XXXXXXX E/OneSignal: Unknown error getting FCM Token
    java.lang.IllegalStateException: Default FirebaseApp is not initialized in this process XXXXXXX. Make sure to call FirebaseApp.initializeApp(Context) first.
        at com.google.firebase.FirebaseApp.getInstance(com.google.firebase:firebase-common@@16.0.2:240)
        at com.google.firebase.iid.FirebaseInstanceId.getInstance(Unknown Source:1)
        at com.google.firebase.iid.zzaw.<init>(Unknown Source:16)
        at com.google.firebase.iid.zzaw.<init>(Unknown Source:1)
        at com.google.firebase.iid.FirebaseInstanceId.<init>(Unknown Source:15)
        at com.google.firebase.iid.FirebaseInstanceId.<init>(Unknown Source:7)
        at com.google.firebase.iid.zzao.create(Unknown Source:3)
        at com.google.firebase.components.zzj.zza(com.google.firebase:firebase-common@@16.0.2:43)
        at com.google.firebase.components.zzk.get(Unknown Source:4)
        at com.google.firebase.components.zzj.get(com.google.firebase:firebase-common@@16.0.2:54)
        at com.google.firebase.components.zza.get(com.google.firebase:firebase-common@@16.0.2:26)
        at com.google.firebase.components.zzf.get(com.google.firebase:firebase-common@@16.0.2:34)
        at com.google.firebase.components.zzf.zza(com.google.firebase:firebase-common@@16.0.2:82)
        at com.google.firebase.FirebaseApp.zze(com.google.firebase:firebase-common@@16.0.2:761)
        at com.google.firebase.FirebaseApp.initializeApp(com.google.firebase:firebase-common@@16.0.2:359)
        at com.onesignal.PushRegistratorFCM.initFirebaseApp(PushRegistratorFCM.java:162)
        at com.onesignal.PushRegistratorFCM.getToken(PushRegistratorFCM.java:101)
        at com.onesignal.PushRegistratorAbstractGoogle.attemptRegistration(PushRegistratorAbstractGoogle.java:97)
        at com.onesignal.PushRegistratorAbstractGoogle.access$100(PushRegistratorAbstractGoogle.java:37)
        at com.onesignal.PushRegistratorAbstractGoogle$1.run(PushRegistratorAbstractGoogle.java:84)
        at java.lang.Thread.run(Thread.java:923)
2021-11-30 10:08:10.368 18827-18986/XXXXXXX I/OneSignal: scheduleSyncServiceAsJob:result: 1
2021-11-30 10:08:13.617 18827-19067/XXXXXXX E/OneSignal: Unknown error getting FCM Token
    java.lang.IllegalStateException: FirebaseApp name ONESIGNAL_SDK_FCM_APP_NAME already exists!
        at com.google.android.gms.common.internal.Preconditions.checkState(Unknown Source:29)
        at com.google.firebase.FirebaseApp.initializeApp(com.google.firebase:firebase-common@@16.0.2:350)
        at com.onesignal.PushRegistratorFCM.initFirebaseApp(PushRegistratorFCM.java:162)
        at com.onesignal.PushRegistratorFCM.getToken(PushRegistratorFCM.java:101)
        at com.onesignal.PushRegistratorAbstractGoogle.attemptRegistration(PushRegistratorAbstractGoogle.java:97)
        at com.onesignal.PushRegistratorAbstractGoogle.access$100(PushRegistratorAbstractGoogle.java:37)
        at com.onesignal.PushRegistratorAbstractGoogle$1.run(PushRegistratorAbstractGoogle.java:84)
        at java.lang.Thread.run(Thread.java:923)
2021-11-30 10:08:13.626 18827-18827/XXXXXXX I/OneSignal: scheduleSyncServiceAsJob:result: 1
2021-11-30 10:08:19.272 18827-19073/XXXXXXX I/OneSignal: Device registered, UserId = ec2f95ce-4fec-4e87-946b-4a282aa7aa51
2021-11-30 10:11:42.034 18827-19224/XXXXXXX I/OneSignal: Error updating the user record because of the null user id

**Comment by jkasten2** [comment]
Thanks for the text log. This looks to be the root issue.
```java
java.lang.IllegalStateException: Default FirebaseApp is not initialized in this process XXXXXXX. Make sure to call FirebaseApp.initializeApp(Context) first.
at com.google.firebase.FirebaseApp.getInstance(com.google.firebase:firebase-common@@16.0.2:240)
...
at com.onesignal.PushRegistratorFCM.initFirebaseApp(PushRegistratorFCM.java:162)
```

Firebase doesn't require a default app so it isn't clear why this error is happening. It looks like it using [`firebase-common:16.0.2`](https://mvnrepository.com/artifact/com.google.firebase/firebase-common/16.0.2) which is from 2018. You can start by trying to use [`firebase-common:16.0.4`](https://mvnrepository.com/artifact/com.google.firebase/firebase-common/16.0.4) which they may have fixed the issue. Or you might have to use even a never version.

**Comment by grimaldistudio** [comment]
Sorry for the delay in replying, but I had to do several tests.

Your indication seems correct, **but despite the update to version 16.0.4 and 16.1.0 I still have the same error**.

If I try to update to version 17+ the compiler fails to build.

I don't know if it can help but **on a device with Android 7 I don't have this problem** (SM-G920F (7.0)).
The error persists with Android 9-10-11. 

2021-11-30 17:20:37.540 727-1443/XXXXXXX E/OneSignal: Unknown error getting FCM Token
    java.lang.IllegalStateException: Default FirebaseApp is not initialized in this process XXXXXXX. Make sure to call FirebaseApp.initializeApp(Context) first.
        at com.google.firebase.FirebaseApp.getInstance(com.google.firebase:firebase-common@@16.1.0:245)
        at com.google.firebase.iid.FirebaseInstanceId.getInstance(Unknown Source:1)
        at com.google.firebase.iid.zzaw.<init>(Unknown Source:16)
        at com.google.firebase.iid.zzaw.<init>(Unknown Source:1)
        at com.google.firebase.iid.FirebaseInstanceId.<init>(Unknown Source:15)
        at com.google.firebase.iid.FirebaseInstanceId.<init>(Unknown Source:7)
        at com.google.firebase.iid.zzao.create(Unknown Source:3)
        at com.google.firebase.components.ComponentRuntime.lambda$new$0(com.google.firebase:firebase-common@@16.1.0:66)
        at com.google.firebase.components.ComponentRuntime$$Lambda$1.get(Unknown Source:4)
        at com.google.firebase.components.Lazy.get(com.google.firebase:firebase-common@@16.1.0:53)
        at com.google.firebase.components.ComponentRuntime.initializeEagerComponents(com.google.firebase:firebase-common@@16.1.0:155)
        at com.google.firebase.FirebaseApp.initializeAllApis(com.google.firebase:firebase-common@@16.1.0:779)
        at com.google.firebase.FirebaseApp.initializeApp(com.google.firebase:firebase-common@@16.1.0:369)
        at com.onesignal.PushRegistratorFCM.initFirebaseApp(PushRegistratorFCM.java:162)
        at com.onesignal.PushRegistratorFCM.getToken(PushRegistratorFCM.java:101)
        at com.onesignal.PushRegistratorAbstractGoogle.attemptRegistration(PushRegistratorAbstractGoogle.java:97)
        at com.onesignal.PushRegistratorAbstractGoogle.access$100(PushRegistratorAbstractGoogle.java:37)
        at com.onesignal.PushRegistratorAbstractGoogle$1.run(PushRegistratorAbstractGoogle.java:84)
        at java.lang.Thread.run(Thread.java:923)
2021-11-30 17:20:42.378 727-1573/XXXXXXX E/OneSignal: Unknown error getting FCM Token
    java.lang.IllegalStateException: FirebaseApp name ONESIGNAL_SDK_FCM_APP_NAME already exists!
        at com.google.android.gms.common.internal.Preconditions.checkState(Unknown Source:29)
        at com.google.firebase.FirebaseApp.initializeApp(com.google.firebase:firebase-common@@16.1.0:360)
        at com.onesignal.PushRegistratorFCM.initFirebaseApp(PushRegistratorFCM.java:162)
        at com.onesignal.PushRegistratorFCM.getToken(PushRegistratorFCM.java:101)
        at com.onesignal.PushRegistratorAbstractGoogle.attemptRegistration(PushRegistratorAbstractGoogle.java:97)
        at com.onesignal.PushRegistratorAbstractGoogle.access$100(PushRegistratorAbstractGoogle.java:37)
        at com.onesignal.PushRegistratorAbstractGoogle$1.run(PushRegistratorAbstractGoogle.java:84)
        at java.lang.Thread.run(Thread.java:923)

**Comment by jkasten2** [comment]
@grimaldistudio I recommend trying to still upgrade to the OneSignal-Cordova-SDK 3.x.x version as these older Google / Firebase libraries might have bugs. Also these old versions will only work so long before the Google Pay store requires you to update them, such as requiring a new `targetSdkVersion`.

Almost all plugins that are still using the "Android Support Library" should automatically work with AndroidX. This is because Google has a tool called Jetifier that automatically runs to upgrade them at build time. If you have issues still with the other plugins I recommend listing the plugins and versions here as well as reporting the issue to those developer plugins on their Github or through the directly.

**Comment by grimaldistudio** [comment]
@jkasten2 Thanks for your assistance, your help has allowed me to shorten the time to identify the problem.
Ultimately, yes, I also think it is necessary to do a general upgrade of the application.

**Comment by chacabuk** [comment]
I have same issue


```
Ionic:

   Ionic CLI                     : 6.18.0 (C:\Users\Pablo\AppData\Roaming\npm\node_modules\@ionic\cli)
   Ionic Framework               : @ionic/angular 6.0.1
   @angular-devkit/build-angular : 13.0.4
   @angular-devkit/schematics    : 13.0.4
   @angular/cli                  : 13.0.4
   @ionic/angular-toolkit        : 5.0.3

Cordova:

   Cordova CLI       : 10.0.0
   Cordova Platforms : android 9.1.0
   Cordova Plugins   : cordova-plugin-ionic-keyboard 2.2.0, cordova-plugin-ionic-webview 4.2.1, (and 8 other plugins)

Utility:

   cordova-res (update available: 0.15.4) : 0.15.2
   native-run (update available: 1.5.0)   : 0.2.8

System:

   Android SDK Tools : 26.1.1 (C:\Users\Pablo\AppData\Local\Android\Sdk\)
   NodeJS            : v16.10.0 (C:\Program Files\nodejs\node.exe)
   npm               : 8.1.2
   OS                : Windows 10
```

```
cordova-plugin-advanced-http 3.2.2 "Advanced HTTP plugin"
cordova-plugin-background-geolocation 3.1.0 "CDVBackgroundGeolocation"
cordova-plugin-device 2.0.2 "Device"
cordova-plugin-file 6.0.2 "File"
cordova-plugin-ionic-keyboard 2.2.0 "cordova-plugin-ionic-keyboard"
cordova-plugin-ionic-webview 4.2.1 "cordova-plugin-ionic-webview"
cordova-plugin-splashscreen 5.0.2 "Splashscreen"
cordova-plugin-statusbar 2.4.2 "StatusBar"
cordova-plugin-whitelist 1.3.3 "Whitelist"
onesignal-cordova-plugin 2.11.4 "OneSignal Push Notifications"
phonegap-plugin-barcodescanner 8.1.0 "BarcodeScanner"
```

`    <preference name="android-targetSdkVersion" value="30" />`

![image](https://user-images.githubusercontent.com/17498840/146589187-d033b50f-bdfb-4ac2-b5b0-74ba7661c140.png)

**Comment by chacabuk** [comment]
I thought I had the same problem, but the PlayerID I was evaluating was the one I got in a previous test.
Check if you are evaluating the correct playerId.

**Comment by grimaldistudio** [comment]
It was impossible to do a general update of the app.

I solved it like this:
in the build.gradle file inside the buildscript block I have added classpath 'com.google.gms: google-services: 4.0.1'
This was the winning move because it solved the error Default FirebaseApp is not initialized in this process ...

`buildscript {
...
   dependencies {
    ...
    classpath 'com.google.gms:google-services:4.0.1'
   }
}`


in the build.gradle file inside the dependencies I added the compile line "com.google.firebase: firebase-common: 16.1.0"
`compile "com.google.firebase: firebase-common: 16.1.0"`

Finally on the Onesignal panel I had to delete the device.
The device was subsequently engaged without errors, now it receives notifications.

I know the solution is very specific but I hope it helps anyway.
